test/wait.go

```diff
@@ -82,17 +83,28 @@ func pollImmediateWithContext(ctx context.Context, fn func() (bool, error)) erro
 // interval until inState returns `true` indicating it is done, returns an
 // error or timeout. desc will be used to name the metric that is emitted to
 // track how long it took for name to get into the state checked by inState.
-func WaitForTaskRunState(ctx context.Context, c *clients, name string, inState ConditionAccessorFn, desc string) error {
+// version will be used to determine the client to be applied for the wait.
+func WaitForTaskRunState(ctx context.Context, c *clients, name string, inState ConditionAccessorFn, desc, version string) error {
 	metricName := fmt.Sprintf("WaitForTaskRunState/%s/%s", name, desc)
 	_, span := trace.StartSpan(context.Background(), metricName)
 	defer span.End()
 
 	return pollImmediateWithContext(ctx, func() (bool, error) {
-		r, err := c.V1beta1TaskRunClient.Get(ctx, name, metav1.GetOptions{})
-		if err != nil {
-			return true, err
+		switch version {
+		case "v1":
+			r, err := c.V1TaskRunClient.Get(ctx, name, metav1.GetOptions{})
+			if err != nil {
+				return true, err
+			}
+			return inState(&r.Status)
+		// TODO: default version needs to be changed to v1 once swapped to stored version
```

Comment:
I don't think this TODO is necessary (here or below)-- we'll probably remove v1beta1 support long before adding a v2 or other version